### PR TITLE
Implement left arrow to focus parent, fix list style on firefox

### LINKED ISSUE #42751: Navigation: Fix focus ring visible on hover after navigating through keyboard
**What happened**:

This ticket needs to be implemented after this [PR](https://github.com/grafana/grafana/pull/41618) is merged.

After navigating using the keyboard, the focus ring became visible in hover interactions. 

https://user-images.githubusercontent.com/239999/144630321-861fbf07-583a-4daa-ac6b-0d6508890c4f.mp4

### LINKED ISSUE #42760: Navigation: Implement Arrow left keyboard navigation in sub item to focus on parent element
**What happened**:

This ticket needs to be implemented after this [PR](https://github.com/grafana/grafana/pull/41618) is merged.

##### Current Behaviour

- Navigating via the keyboard on navBar Items opens a flyout menu 
- when pressing the "Arrow Right" key the focus move the first item of the submenu, 

##### Missing Implementation

- Pressing "Arrow left" should focus on the parent element without closing the sub-menu. 

arrowLeft

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- implements left arrow to focus parent item
- stops showing bullet points in firefox when `newNavigation` toggle is enabled:
  - | before | after |
    | --- | --- |
    | image | image |
- fixes focus styles showing when interacting with mouse

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/42760
Fixes https://github.com/grafana/grafana/issues/42751

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/components/NavBar/NavBarItemMenuItem.tsx [hunk 1]
```diff
@@ -45,8 +45,17 @@ export function NavBarItemMenuItem({ item, state, onNavigate }: NavBarItemMenuIt
     ref
   );
 
+  const { keyboardProps } = useKeyboard({
+    onKeyDown: (e) => {
+      if (e.key === 'ArrowLeft') {
+        onClose();
```
Comment (kaydelaney) [inline public/app/core/components/NavBar/NavBarItemMenuItem.tsx:51]: I don't think it really matters in this case, but should we add `preventDefault` here for "correctness"?

**Review by kaydelaney — APPROVED** [review]
Looks good! Just left one small comment.
I do think that perhaps the arrows keys should be use to move between top-level navigation items rather than tabbing through them one-by-one, but that's a separate issue. Nice work!

**Review by hugohaggmark — COMMENTED** [review]
I don't think this was the intended (ping @axelavargas) behaviour but the code looks good. Can we add a test case for this?

File: public/app/core/components/NavBar/NavBarItemMenuItem.tsx (hunk 1 above)
Comment (axelavargas) [reply public/app/core/components/NavBar/NavBarItemMenuItem.tsx:51]: btw @kaydelaney I think in this case is not needed to add the `prevenDefault`, we are using `useKeyboard` and the hook is doing it automatically.

**Review by axelavargas — COMMENTED** [review]
@ashharrison90 great first attempt! I know this is the easiest way to achieve moving the focus to the parent, but do you think is there another way we could achieve the original intention?

**Review by axelavargas — APPROVED** [review]
👏🏾 👏🏾 really cool approach Mr. Harrisson 😎  works as expected, I think we can an extra test case in the  https://github.com/grafana/grafana/blob/main/public/app/core/components/NavBar/NavBarItem.test.tsx#L134

**Pull request merged**